WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -87,6 +87,9 @@ public static GolombRiceFilter CreateDummyEmptyFilter(uint256 blockHash)
 
 		public void Synchronize()
 		{
+			// Check permissions.
+			using var _ = File.Open(IndexFilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite);
```

Comment:
I would prefer to do not catch the `System.UnauthorizedAccessException` exception instead. I mean, instead of:

```c#
catch (Exception ex){
     Logger.LogDe...
}
``` 

I would prefer:

```c#
catch (UnauthorizedAccessException)
{
    throw;
}
catch (Exception ex){
     Logger.LogDe...
}
```

Reply:
I am not sure. Nothing can stop the while loop so far (only intentional stop), this could be the first - that could mess up the logic there. I would not do that. Also, I don't know what happens after the throw - nobody will see it on the server and there is no recovery mechanism.

Reply:
ummm okay.